### ISSUE TITLE
Couchbase integration test parse error

### ISSUE THREAD

**Issue by alissapajer** [quasar-analytics/quasar#2114]
I've seen this sort of error pop up occasionally:
```
[error] ! match like complement [./matchLikeComplement.test] (6 seconds, 393 ms)
[error]  java.lang.RuntimeException: Failed to read datum: reason='ParseError(JSON terminates unexpectedly.)', datum= (rethrow.scala:33)
[error] quasar.rethrow$$anon$1.quasar$rethrow$$anon$1$$$anonfun$1(rethrow.scala:33)
```
Unfortunately that's all it prints out.

That test was from `quasar.ViewReadQueryRegressionSpec` run in the `couchbase` travis build. I have not seen the error outside of travis, but I also rarely run all the CB integration tests locally.

**Comment by alissapajer** [comment]
Different test; same error. This particular test seems to throw this error more frequently than others, from the few times I've seen this. Again this was in `quasar.ViewReadQueryRegressionSpec`.
```
[error] ! project from non-static concatted array [./projectFromConcattedArray.test] (5 seconds, 550 ms)
[error]  java.lang.RuntimeException: Failed to read datum: reason='ParseError(JSON terminates unexpectedly.)', datum= (rethrow.scala:33)
[error] quasar.rethrow$$anon$1.quasar$rethrow$$anon$1$$$anonfun$1(rethrow.scala:33)
```

**Comment by drostron** [comment]
This seems to occur more often when the Travis build cluster is under load. If that's the case this could be one of the CB server errors that's being dropped (#1621 ) and therefore we're attempting to parse results incorrectly. If so, we'll see something more meaningful once #2112 lands.

**Comment by drostron** [comment]
This error is still occurring after the #2112 addition. Upon further investigation with @nicflores, we discovered at least one case where the CB client is consistently offering invalid JSON for a `N1qlQueryRow`. It appears this is also what's happening intermittently on Travis, yet the cause of its intermittent nature is still unclear.

SQL²:
```
select address from `/cb/beer-sample/brewery`
```

Fails with:
```
Quasar error: Failed to read datum: reason='ParseError(JSON terminates unexpectedly.)', datum=[
            "563 Second Street"
```

N1QL:
```
select value v from (select value (case when isobject(`_10`) then `_10`.['address'] else null end) from (select value ifmissing(`_9`.['value'], `_9`) from `beer-sample` as `_9` where (`type` = 'brewery')) as `_10`) v
```

The N1QL query returns expected results when issued from `cbq`.

**Comment by nicflores** [comment]
Fixed by #2130. Please re-open if you see this again.
